graphix/sim/base_backend.py

```diff
@@ -1,14 +1,21 @@
 from __future__ import annotations
 
+import numbers
+
 import numpy as np
 
 import graphix.clifford
 import graphix.command
+import graphix.parameter
 import graphix.pauli
 
 
 def op_mat_from_result(vec: tuple[float, float, float], result: bool) -> np.ndarray:
-    op_mat = np.eye(2, dtype=np.complex128) / 2
+    if all(isinstance(value, numbers.Number) for value in vec):
+        op_mat = np.eye(2, dtype=np.complex128) / 2
+    else:
+        # At least one value is symbolic (i.e., parameterized).
+        op_mat = np.eye(2, dtype="O") / 2
```

Comment:
`dtype="O"` needs extra care as it can impact the performance of simulators.

Reply:
We know that symbolic simulations are extra-costly!

Reply:
Then I suggest completely isolating the logic to different functions: `numpy` is notorious for unpredictable automatic type conversion, so I don't want it close to numerical simulator that should be as lightweight as possible.
Current approach (e.g., add conditional branches to existing functions) is very vulnerable to accidental contamination.